tests/MCPClient/test_policy_check.py

```diff
@@ -0,0 +1,592 @@
+def derivation_for_access(
+    sip_file: models.File, access_file: models.File
+) -> models.Derivation:
+    return models.Derivation.objects.create(
+        source_file=sip_file, derived_file=access_file
+    )
+
+
+@pytest.fixture
+def access_file_format_version(
+    access_file: models.File, format_version: fprmodels.FormatVersion
+) -> models.FileFormatVersion:
+    return models.FileFormatVersion.objects.create(
+        file_uuid=access_file, format_version=format_version
+    )
+
+
+STDOUT = json.dumps(
+    {"eventOutcomeInformation": "pass", "eventOutcomeDetailNote": "a note"}
+)
+
+
+@pytest.fixture
+def EXECUTE_OR_RUN() -> Any:
+    with mock.patch(
+        "policy_check.executeOrRun",
+        return_value=(
+            0,
+            STDOUT,
+            "",
+        ),
+    ):
+        yield
+
+
+@pytest.mark.django_db
+@mock.patch("policy_check.executeOrRun")
+def test_policy_check_succeeds_if_rules_exist(
+    execute_or_run: mock.Mock,
+    sip_file: models.File,
+    sip: models.SIP,
+    fprule_policy_check: fprmodels.FPRule,
+    sip_file_format_version: models.FileFormatVersion,
+    format: fprmodels.Format,
+    format_version: fprmodels.FormatVersion,
+    shared_directory_path: str,
+) -> None:
+    job = mock.Mock(
+        args=[
+            "policy_check",
+            sip_file.currentlocation.decode(),
+            str(sip_file.uuid),
+            str(sip.uuid),
+            str(shared_directory_path),
+            sip_file.filegrpuse,
+        ],
+        JobContext=mock.MagicMock(),
+        spec=Job,
+    )
+    expected_stdout = json.dumps(
+        {
+            "eventOutcomeInformation": "pass",
+            "eventOutcomeDetailNote": f'format="{format.description}"; version="{format_version.version}"; result="Well-Formed and valid"',
+        }
+    )
+    execute_or_run.return_value = (0, expected_stdout, "")
+
+    policy_check.call([job])
+
+    job.set_status.assert_called_once_with(0)
+    assert (
+        models.Event.objects.filter(
+            file_uuid_id=sip_file.uuid,
+            event_type="validation",
+            event_detail=(
+                f'program="{fprule_policy_check.command.tool.description}";'
+                f' version="{fprule_policy_check.command.tool.version}"'
+            ),
+            event_outcome="pass",
+            event_outcome_detail=json.loads(expected_stdout)["eventOutcomeDetailNote"],
+        ).count()
+        == 1
+    )
+    assert job.pyprint.mock_calls == [
+        mock.call("Running", fprule_policy_check.command.description),
+        mock.call(
+            f"Command {fprule_policy_check.command.description} completed with output {expected_stdout}"
+        ),
+        mock.call(
+            f"Creating policy checking event for {sip_file.currentlocation.decode()} ({sip_file.uuid})"
+        ),
+    ]
+
+
+@pytest.mark.django_db
+@mock.patch(
+    "policy_check.executeOrRun",
+    return_value=(
+        1,
+        STDOUT,
+        "",
+    ),
```

Comment:
I think several of these `return_value` tuples can be represented in a single line if you remove the trailing comma.